controllers/vspheremachine_controller.go

```diff
@@ -317,9 +317,9 @@ func (r machineReconciler) reconcileNormal(ctx *context.MachineContext) (reconci
 	ctrlutil.AddFinalizer(ctx.VSphereMachine, infrav1.MachineFinalizer)
 
 	if !ctx.Cluster.Status.InfrastructureReady {
-		ctx.Logger.Info("Cluster infrastructure is not ready yet")
+		ctx.Logger.Info("Cluster infrastructure is not ready yet, requeuing")
 		conditions.MarkFalse(ctx.VSphereMachine, infrav1.VMProvisionedCondition, infrav1.WaitingForClusterInfrastructureReason, clusterv1.ConditionSeverityInfo, "")
-		return reconcile.Result{}, nil
+		return reconcile.Result{RequeueAfter: 30*time.Second}, nil
```

Comment:
Should we have a watch here instead of requeueing?

Reply:
@vincepri oh you mean just like there's a watch for Cluster for the VSphereCluster object, we should add one mapping Cluster to VSphereMachine?

Reply:
Yeah if we're waiting for it, we probably should add watch instead of using RequeueAfter

Reply:
oh looks like there is a watch for Cluster, just need to update the predicate to check for this

Reply:
I tested changing the update predicate for the watch on Cluster but I see the same, machines reconcile after a while. I'll see if I missed something and test again

Reply:
The watch on cluster object's infrastructureReady field change wasn't working because it wasn't enqueuing the vsphereMachine object. Opened a PR for that https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1306
With the change in clusterToVsphereMachine mapper, updating the watch on cluster object to enqueue vsphereMachine when cluster.status.infrastructureReady becomes true works. I'll rebase this PR when the other one is merged